docs/inference/README.md

```diff
@@ -55,10 +62,34 @@ inference:
         name: "falcon-7b-adapter"
         image:  "<YOUR_IMAGE>"
       strength: "0.2"
-    - source:
-        name: "additional-source"
-        image: "<YOUR_ADDITIONAL_IMAGE>"
-      strength: "0.5" 
 ```
+Currently, only images are supported as adapter sources. The `strength` field specifies the multiplier for applying the adapter weights to the raw model weights.
+
+Note: if you build a container image for an existing adapter, please copy all adapter files to the **/data** directory inside the container.
+
+The detailed `InferenceSpec` API definitions can be found [here](https://github.com/Azure/kaito/blob/2ccc93daf9d5385649f3f219ff131ee7c9c47f3e/api/v1alpha1/workspace_types.go#L75).
+
+
+# Inference workload
+
+Depending on whether the specified model supports distributed inference or not, the Kaito controller will choose to use either Kubernetes **apps.deployment** workload (by default) or Kubernetes **apps.statefulset** workload (if the model supports distributed inference) to manage the inference service, which is exposed using a Cluster-IP type of Kubernetes `service`.
+
+When adapters are specified in the `inference` spec, the Kaito controller will add initcontainers for every adapter in addition to the main container. The pod structure is illustrated in Figure 1.
+
+<div align="left">
+  <img src="../img/kaito-inference-adapter.png" width=40% title="Kaito inference adapter" alt="Kaito inference adapter">
+</div>
+Figure 1. Kaito inference pod (with adapters) structure.
+
+If an image is specified in the adapter source, the corresponding initcontainer uses the specified image as the container image. These initcontainers ensures all adapter data are available locally before the inference service starts. The main container uses one of the supported model images. The image entry launches the [inference\_api.py](../../presets/inference/text-generation/inference_api.py) script.
+
+All containers use shared local volumes (by mounting the same EmptyDir volumes), hence file copies between containers are avoided.
+
+## Workload update
+
+Users can update the `workspace` customer resource and change the `adapters` field in the inference spec. The Kaito controller will apply a workload deployment update to make adapter changes take effect. The inference service pod will be recreated, causing a short period of service downtime. Once the new adapters are merged with the raw model weights and loaded into the GPU memory, the service will resume.
```

Comment:
```suggestion
## Workload update

To update the `adapters` field in the `inference` spec, users can modify the `workspace` custom resource. The Kaito controller will apply the changes, triggering a workload deployment update. This will recreate the inference service pod, resulting in a brief service downtime. Once the new adapters are merged with the raw model weights and loaded into GPU memory, the service will resume.
```